### PR TITLE
doc: Add instructions for utilizing the pre-staging environment

### PR DESCRIPTION
**Pull request by ppawlowski**
## Description

This Pull Request extends development documentation by adding information about the possibility of validating changes before emerging to the `main` branch.

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by joepavitt** [comment]
Why can't we do this for _every_ PR - why does it need the manual step of adding a label?

**Comment by knolleary** [comment]
@joepavitt because so far we choose not to.

I believe that for many PRs the test env would go untouched because the PR has been tested locally and the nature of the change doesn't require a dedicated env for further manual testing. We would be throwing money at AWS for no value in return.

If we decided it was worthwhile spinning up on every PR we could do so. I don't think that's necessary today.

**Comment by hardillb** [comment]
@joepavitt because not every change needs a K8s environment to test it

**Comment by ppawlowski** [comment]
@joepavitt  AFAIR this was the main expectation - to have a possibility to verify changes which may break the staging environment.  Not all of them will tend to do this and can be merged and deployed to staging without validation on the pre-staging environment. 
@ZJvandeWeg as it is mentioned in the docs, `This environment is terminated upon PR merging or closure.`

**Comment by ppawlowski** [comment]
To be more clear - from the infrastructure perspective, an environment is in fact FlowFuse Cloud instance running on an existing EKS cluster we use for the staging environment. No new AWS resources are created.

**Comment by joepavitt** [comment]
Can we just sanity check one thing, fairly sure it's covered by the `synchronise` event, but if the base `main` branch is updated, would we auto-deploy the relevant changes to any open PRs?

Thinking of a situation where we have _potentially_ clashing PRs. When one is merged (having been verified in staging) that would update `main`, an associated PR should have it's staging environment also updated to reflect the latest `main`.

**Comment by joepavitt** [comment]
Let's get 1. and 2. ticked of please @ppawlowski

File: src/handbook/development/contributing.md [hunk 1]
```diff
@@ -61,4 +61,9 @@ When merging a PR, you should choose the "Merge pull request" option. There is n
 
 When conducting a PR review, if you are the last (or only) reviewer and all reviews (including your own) are approvals, unless there is a comment from the author stating otherwise, you are free to conduct the merge. Otherwise, leave the merge to the author of the PR, or a future reviewer.
 
+For a comprehensive review of the Pull Request, utilize the designated FlowFuse pre-staging environment. As of the composition of this document, the pre-staging verification is only available for the primary [FlowFuse NPM package](https://github.com/FlowFuse/flowfuse).
+To validate changes within the pre-staging environment, apply the deploy:pr label to the PR. The PR will then undergo automatic deployment to the pre-staging environment, and the deployment link will be appended to the associated workflow summary. The label should remain until the PR is merged.
```
Comment (joepavitt) [inline src/handbook/development/contributing.md:65]:
```suggestion
To validate changes within the pre-staging environment, apply the `deploy:pr` label to the PR. The PR will then undergo automatic deployment to the pre-staging environment, and the deployment link will be appended to the associated workflow summary. The label should remain until the PR is merged.
```

Comment (joepavitt) [reply src/handbook/development/contributing.md:65]: Can we also get a bit more detail on where to find the link (maybe a screenshot) - given these instructions, and [this example PR](https://github.com/FlowFuse/flowfuse/pull/3652#discussion_r1542676749) - I cannot find the link

**Review by joepavitt — CHANGES_REQUESTED** [review]

**Review by joepavitt — APPROVED** [review]

**Pull request merged**